component/config_forwarding.libsonnet

```diff
@@ -117,69 +137,167 @@ local patchCustomPipelines = {
+    patchLegacyAuditLogDefaults,
+    patchLegacyJsonLogging,
+    patchLegacyMultilineErrors,
+    patchLegacyCustomInputs,
+    patchLegacyCustomOutputs,
+    patchLegacyCustomPipelines,
   ],
   {
     inputs: {},
     outputs: {},
     pipelines: {},
-  }
-);
+  },
+) + com.makeMergeable(params.clusterLogForwarder);
+
+// Unfold objects into array for ClusterLogForwarder resource.
+local unfoldSpecs(specs) = {
+  // Unfold objects into array.
+  [if std.length(specs.inputs) > 0 then 'inputs']: [
+    { name: name } + specs.inputs[name]
+    for name in std.objectFields(specs.inputs)
+  ],
+  [if std.length(specs.outputs) > 0 then 'outputs']: [
+    { name: name } + specs.outputs[name]
+    for name in std.objectFields(specs.outputs)
+  ],
+  [if std.length(specs.pipelines) > 0 then 'pipelines']: [
+    { name: name } + specs.pipelines[name]
+    for name in std.objectFields(specs.pipelines)
+  ],
+} + {
+  // Import remaining specs as is.
+  [key]: specs[key]
+  for key in std.objectFields(specs)
+  if !std.member([ 'inputs', 'outputs', 'pipelines' ], key)
+};
 
 // ClusterLogForwarder:
 // Create definitive ClusterLogForwarder resource from specs.
 local clusterLogForwarder = lib.ClusterLogForwarder(params.namespace, 'instance') {
-  spec: {
-    // Unfold objects into array.
-    [if std.length(clusterLogForwarderSpec.inputs) > 0 then 'inputs']: [
-      { name: name } + clusterLogForwarderSpec.inputs[name]
-      for name in std.objectFields(clusterLogForwarderSpec.inputs)
-    ],
-    [if std.length(clusterLogForwarderSpec.outputs) > 0 then 'outputs']: [
-      { name: name } + clusterLogForwarderSpec.outputs[name]
-      for name in std.objectFields(clusterLogForwarderSpec.outputs)
-    ],
-    [if std.length(clusterLogForwarderSpec.pipelines) > 0 then 'pipelines']: [
-      { name: name } + clusterLogForwarderSpec.pipelines[name]
-      for name in std.objectFields(clusterLogForwarderSpec.pipelines)
-    ],
-  } + {
-    // Import remaining specs as is.
-    [key]: clusterLogForwarderSpec[key]
-    for key in std.objectFields(clusterLogForwarderSpec)
-    if !std.member([ 'inputs', 'outputs', 'pipelines' ], key)
-  },
+  spec: unfoldSpecs(clusterLogForwarderSpec),
 };
 
+// namespaceLogForwarder:
+// Create namespaced LogForwarder resource from specs.
+local namespaceLogForwarder = [
+  local specs = { inputs: {}, outputs: {}, pipelines: {} } + com.makeMergeable(params.namespaceLogForwarder[forwarder]);
+  local name = utils.namespacedName(forwarder).name;
+  local namespace = utils.namespacedName(forwarder).namespace;
+  local serviceAccount = std.get(specs, 'serviceAccountName', utils.namespacedName(forwarder).name);
+
+  lib.ClusterLogForwarder(namespace, name) {
+    spec: { serviceAccountName: serviceAccount } + com.makeMergeable(unfoldSpecs(specs)),
+  }
+  for forwarder in std.objectFields(params.namespaceLogForwarder)
+  if !std.member([ 'enabled', 'instance', 'openshift-logging/instance' ], forwarder)
```

Comment:
I'd recommend moving the namespace log forwarder instances to parameter `params.namespaceLogForwarder.instances` so that we have more wiggle room to add additional top-level parameters for the feature without having to remember to update this list.

Excluding the `openshift-logging/instance` forwarder is a nice touch though.